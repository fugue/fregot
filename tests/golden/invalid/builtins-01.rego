# 'all' and 'any'
package fregot.tests.invalid.builtins_01

test_all_object {
  all({"foo": true, "bar": false, "baz": 100})
}

test_any_object {
  any({"foo": true, "bar": false, "baz": 100})
}
