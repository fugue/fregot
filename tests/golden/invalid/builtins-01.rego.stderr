fregot (typecheck error):
  "builtins-01.rego" (line 5, column 3):
  Subtype error:

    5|   all({"foo": true, "bar": false, "baz": 100})
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  Cannot use type object{
    "bar": boolean,
    "foo": boolean,
    "baz": number
  } as set{boolean}|object{any: boolean}|array{number: boolean}

fregot (typecheck error):
  "builtins-01.rego" (line 9, column 3):
  Subtype error:

    9|   any({"foo": true, "bar": false, "baz": 100})
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  Cannot use type object{
    "bar": boolean,
    "foo": boolean,
    "baz": number
  } as set{boolean}|object{any: boolean}|array{number: boolean}
